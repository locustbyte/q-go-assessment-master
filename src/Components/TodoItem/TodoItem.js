import React from 'react';
import moment from 'moment';

class TodoItem extends React.Component {

    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this, this.props.todo.id);
        this.handleClickDelete = this.handleClickDelete.bind(this, this.props.todo.id);
    }

    handleClick(id) {
        this.props.toggleDone(id);
    }

    handleClickDelete(id) {
        this.props.removeTodo(id);
    }

    render() {
        return (
            <li className={`completed-${this.props.todo.completed}`}>
                <input type="checkbox" onClick={this.handleClick} /><strong>TASK:</strong> {this.props.todo.title} <span className="added">added {moment(this.props.todo.added).fromNow()}</span> <span onClick={this.handleClickDelete} ><i className={'fa fa-minus-circle'} aria-hidden="true"></i> DELETE</span>
            </li>
        );
    };
}

TodoItem.defaultProps = {
  todo: {
      title: "Empty todo",
      added: moment(),
      completed: false
  },
  
};

TodoItem.propTypes = {
  todo: React.PropTypes.object.isRequired,
};

export default TodoItem;
