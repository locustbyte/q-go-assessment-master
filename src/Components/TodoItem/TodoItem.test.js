import React from 'react';
import renderer from 'react-test-renderer';

import ReactDOM from 'react-dom';
import TodoItem from './TodoItem';

import uuid from 'uuid';
import moment from 'moment';

const todo = {
    title: "Call the evangelist",
    completed: true,
    added: moment().toISOString(),
    category: 2,
    id: uuid.v4()
};

const categories = [
    "Private",
    "Work",
    "Free time",
    "School",
    "Shopping"
];

describe('<TodoItem />', () =>{

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TodoItem todo={todo} categories={categories} />, div);
    });

    it('should match its empty snapshot', () => {
        const tree = renderer.create(
            <TodoItem />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render a todo item with a classname', () => {
        const tree = renderer.create(
            <TodoItem todo={todo} categories={categories} ></TodoItem>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should attach an onclick handler passed in', () => {
        const tree = renderer.create(
            <TodoItem todo={todo} categories={categories} onClick={() => {}}></TodoItem>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


    it('should render its children', () => {
        const tree = renderer.create(
            <TodoItem todo={todo} categories={categories}>a child here 👶🏻</TodoItem>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
