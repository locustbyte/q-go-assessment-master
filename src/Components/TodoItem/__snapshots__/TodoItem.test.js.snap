exports[`<TodoItem /> should attach an onclick handler passed in 1`] = `
<li
  className="completed-true">
  <input
    onClick={[Function]}
    type="checkbox" />
  <strong>
    TASK:
  </strong>
   
  Call the evangelist
   
  <span
    className="added">
    added 
    a few seconds ago
  </span>
   
  <span
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="fa fa-minus-circle" />
     DELETE
  </span>
</li>
`;

exports[`<TodoItem /> should match its empty snapshot 1`] = `
<li
  className="completed-false">
  <input
    onClick={[Function]}
    type="checkbox" />
  <strong>
    TASK:
  </strong>
   
  Empty todo
   
  <span
    className="added">
    added 
    a few seconds ago
  </span>
   
  <span
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="fa fa-minus-circle" />
     DELETE
  </span>
</li>
`;

exports[`<TodoItem /> should render a todo item with a classname 1`] = `
<li
  className="completed-true">
  <input
    onClick={[Function]}
    type="checkbox" />
  <strong>
    TASK:
  </strong>
   
  Call the evangelist
   
  <span
    className="added">
    added 
    a few seconds ago
  </span>
   
  <span
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="fa fa-minus-circle" />
     DELETE
  </span>
</li>
`;

exports[`<TodoItem /> should render its children 1`] = `
<li
  className="completed-true">
  <input
    onClick={[Function]}
    type="checkbox" />
  <strong>
    TASK:
  </strong>
   
  Call the evangelist
   
  <span
    className="added">
    added 
    a few seconds ago
  </span>
   
  <span
    onClick={[Function]}>
    <i
      aria-hidden="true"
      className="fa fa-minus-circle" />
     DELETE
  </span>
</li>
`;
